azure/ConsiderationsForServiceDesign.md

```diff
@@ -77,6 +77,65 @@ Understanding how your service is used and defining its model and interaction pa
 
 :white_check_mark: **DO** create an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a good developer experience.
+
+Resources are typically described by nouns. Resource and property names must be descriptive and easy to understand for end-users. Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
+### Recommended Naming Conventions
+
+The following are recommended naming conventions for Azure services:
+
+:heavy_check_mark: **DO** use singular nouns for resource (schema) names.
+
+:heavy_check_mark: **DO** use plural nouns for collections (for listing type or collection properties).
+
+:ballot_box_with_check: **YOU SHOULD** case all acronyms the same as a normal word.
+
+For example, `nextUrl` and not `nextURL`.
+
+:ballot_box_with_check: **YOU SHOULD** use "at" suffix in names of `date-time` values.
```

Comment:
This is in direct conflict with earlier naming guidance in the Microsoft REST API guidelines which stated a type suffix for date time typed properties. Just so we are in agreement that value of the the new/changed guideline outweigh the cost of the change.

Reply:
That's true and I meant to make this same exact comment on my own PR. I wrote it this way because this is the guidance I've heard us give in past API reviews. But we should change to the Microsoft (and Graph) convention here unless there is good cause to "go our own way".

Reply:
I did an analysis of the API definitions currently in the repo and found that the `DateTime` suffix is more common than `At` -- and there were several other flavors but all less common than `DateTime`. So I will switch the guidance to that.